runtime/shared_common/shrinit.cpp

```diff
@@ -3070,6 +3070,17 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 	UnitTest::unitTest = UnitTest::NO_TEST;
 	vm->sharedClassConfig = NULL;
 
+	if (FALSE == vm->sharedCacheAPI->xShareClassesPresent) {
+		Trc_SHR_Assert_True(J9_SHARED_CACHE_DEFAULT_BOOT_SHARING(vm));
+		Trc_SHR_Assert_True(J9_ARE_ALL_BITS_SET(runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHEBOOTCLASSES));
+		Trc_SHR_Assert_True(J9_ARE_ALL_BITS_SET(runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_NONFATAL));
+		Trc_SHR_Assert_True(J9_ARE_ALL_BITS_SET(runtimeFlags, J9SHR_RUNTIMEFLAG_DISABLE_CORRUPT_CACHE_DUMPS));
+		Trc_SHR_Assert_True(0 == vm->sharedCacheAPI->verboseFlags);
+		if (J9_ARE_ALL_BITS_SET(vm->verboseLevel, VERBOSE_SHAREDCLASSES)) {
```

Comment:
This seems limiting, in that the other verbose flags cannot be specified for bootstrap only sharing. Similarly for the other shared cache options. I suggest adding a onlyBootclasspath option (or similar, but there is already a noBootclasspath option). This allows all the shared cache options to be used in combination with this.

Reply:
I can add an option to share bootstrap classes only.